core/src/test/java/com/linecorp/armeria/server/HttpServerHandlerTest.java

```diff
@@ -0,0 +1,100 @@
+ *
+ *   https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.awaitility.Awaitility.await;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.linecorp.armeria.client.HttpClient;
+import com.linecorp.armeria.common.AggregatedHttpResponse;
+import com.linecorp.armeria.common.HttpResponse;
+import com.linecorp.armeria.common.HttpStatus;
+import com.linecorp.armeria.common.logging.RequestLog;
+import com.linecorp.armeria.testing.junit.server.ServerExtension;
+
+class HttpServerHandlerTest {
+
+    private static final AtomicReference<RequestLog> logHolder = new AtomicReference<>();
+
+    @RegisterExtension
+    static final ServerExtension server = new ServerExtension() {
+        @Override
+        protected void configure(ServerBuilder sb) throws Exception {
+            sb.accessLogWriter(logHolder::set, true);
+            sb.route()
+              .get("/hello")
+              .build((ctx, req) -> HttpResponse.of(200));
+            sb.route()
+              .get("/httpStatusException")
+              .build((ctx, req) -> {
+                  throw HttpStatusException.of(201);
+              });
+            sb.route()
+              .get("/httpResponseException")
+              .build((ctx, req) -> {
+                  throw HttpResponseException.of(201);
+              });
+        }
+    };
+
+    @BeforeEach
+    void clearLog() {
+        logHolder.set(null);
+    }
+
+    @Test
+    void methodNotAllowed() {
+        final HttpClient client = HttpClient.of(server.uri("/"));
+        final AggregatedHttpResponse res = client.delete("/hello").aggregate().join();
+        assertThat(res.status()).isSameAs(HttpStatus.METHOD_NOT_ALLOWED);
+        await().untilAsserted(() -> assertThat(logHolder.get().path()).isEqualTo("/hello"));
+        assertThat(logHolder.get().requestCause()).isExactlyInstanceOf(HttpStatusException.class);
+    }
+
+    @Test
+    void handleNonExistentMapping() {
+        final HttpClient client = HttpClient.of(server.uri("/"));
+        final AggregatedHttpResponse res = client.get("/non_existent").aggregate().join();
+        assertThat(res.status()).isSameAs(HttpStatus.NOT_FOUND);
+        await().untilAsserted(() -> assertThat(logHolder.get().path()).isEqualTo("/non_existent"));
+        assertThat(logHolder.get().requestCause()).isNull();
+    }
+
+    @Test
+    void httpStatusException() {
```

Comment:
I found out that we leave the `httpStatusException` as the cause but not for `httpResponseException`.
I think we should treat them in the same way and I think we do not log the causes in both cases.

Reply:
Agreed. Could you update accordingly?

Reply:
Fixed. PTAL 🙇